library/lib_location_constants.ks

```diff
@@ -1,34 +1,78 @@
 
   // horizontal landing locations
-  "runway_09_start", Kerbin:GeoPositionLatLng(-0.0487, -74.7247),
-  "runway_09_end", Kerbin:GeoPositionLatLng(-0.0501, -74.4880),  // runway "lip"
-  "runway_27_start", Kerbin:GeoPositionLatLng(-0.0501, -74.4925),
-  "runway_27_end", Kerbin:GeoPositionLatLng(-0.0487, -74.7292),  // runway "lip"
-  "l1_runway_09_start", Kerbin:GeoPositionLatLng(-.0489, -74.7101),
-  "l1_runway_27_start", Kerbin:GeoPositionLatLng(-.0503, -74.5076),
-  "l2_runway_09_start", Kerbin:GeoPositionLatLng(-.0486, -74.7134),
-  "l2_runway_27_start", Kerbin:GeoPositionLatLng(-.0501, -74.5046),
-  "island_runway_09_start", Kerbin:GeoPositionLatLng(-1.5177, -71.9663),
-  "island_runway_27_start", Kerbin:GeoPositionLatLng(-1.5158, -71.8524),
-  "desert_runway_36_start", Kerbin:GeoPositionLatLng(-6.5998, -144.0407),
-  "desert_runway_18_start", Kerbin:GeoPositionLatLng(-6.4480, -144.0383)
-).
+  kerbinLocations:add("runway_09_start", Kerbin:GeoPositionLatLng(-0.0486, -74.7247)).
+  kerbinLocations:add("runway_09_overrun", Kerbin:GeoPositionLatLng(-0.0502, -74.4880)).  // runway "lip"
+  kerbinLocations:add("runway_27_start", Kerbin:GeoPositionLatLng(-0.0502, -74.4925)).
+  kerbinLocations:add("runway_27_overrun", Kerbin:GeoPositionLatLng(-0.0486, -74.7292)).  // runway "lip"
+  kerbinLocations:add("l1_runway_09_start", Kerbin:GeoPositionLatLng(-.0489, -74.7101)).
+  kerbinLocations:add("l1_runway_27_start", Kerbin:GeoPositionLatLng(-.0501, -74.5076)).
+  kerbinLocations:add("l2_runway_09_start", Kerbin:GeoPositionLatLng(-.0486, -74.7134)).
+  kerbinLocations:add("l2_runway_27_start", Kerbin:GeoPositionLatLng(-.0501, -74.5046)).
+  kerbinLocations:add("island_runway_09_start", Kerbin:GeoPositionLatLng(-1.5177, -71.9663)).
+  kerbinLocations:add("island_runway_27_start", Kerbin:GeoPositionLatLng(-1.5158, -71.8524)).
+  kerbinLocations:add("desert_runway_36_start", Kerbin:GeoPositionLatLng(-6.5998, -144.0409)).
+  kerbinLocations:add("desert_runway_18_start", Kerbin:GeoPositionLatLng(-6.4480, -144.0383)).
+
+  location_constants:add("runway_start",kerbinLocations["runway_09_start"]).
+  location_constants:add("reverse_runway_start",kerbinLocations["runway_27_start"]).
+
+  location_constants:add("kerbin",kerbinLocations).
+}
 
 // aliases
-set LocationConstants["runway_start"] to LocationConstants["runway_09_start"].
-set LocationConstants["reverse_runway_start"] to LocationConstants["runway_27_start"].
-
-for key in LocationConstants:keys {
-  if key:contains("desert") {
-    local alias to key:replace("desert", "dessert").
-    set LocationConstants[alias] to LocationConstants[key].
-  }.
-}.
+until not aliasing(location_constants).
+
+local function aliasing {
+  parameter locationConstants.
+  local didChange is false.
+  for key in locationConstants:keys {
+    local bodyLex is locationConstants[key].
+    if bodyLex:istype("lexicon") {
+      local bodyKeys is bodyLex:keys:copy.
```

Comment:
`:copy` is redundant. `:keys` already creates a new list and populates it. (I did an experiment where I saved `:keys` in a variable and then added a new element to the lexicon. The saved list remained unchanged.)

Reply:
While the copy is redundant I used it because I don't like having the thing being changed in the `FOR` loop directly referenced in setting up said loop.  Mostly because for all other collections if you tried to do that kOS would crash when you tried to modify the collection.  I did not set up a copy for the base lexicon's keys because that lexicon does not get modified only it's continence does.  I was surprised that the `FOR` you had worked with no problems when I first saw it.